allennlp/data/dataset_readers/snli.py

```diff
@@ -68,16 +68,23 @@ def text_to_instance(
     ) -> Instance:
 
         fields: Dict[str, Field] = {}
-        premise_tokens = self._tokenizer.tokenize(premise)
-        hypothesis_tokens = self._tokenizer.tokenize(hypothesis)
-        fields["premise"] = TextField(premise_tokens, self._token_indexers)
-        fields["hypothesis"] = TextField(hypothesis_tokens, self._token_indexers)
+
+        if isinstance(self._tokenizer, PretrainedTransformerTokenizer):
```

Comment:
Slight preference for having this be a configuration option instead of relying on this type check.